tidy3d/components/geometry.py

```diff
@@ -3280,8 +3289,374 @@ def _surface_area(self, bounds: Bound) -> float:
+    def _check_mesh(cls, val: SurfaceMeshDataset) -> SurfaceMeshDataset:
+        """Check that the mesh is valid."""
+        if val is None:
+            return None
+        mesh = cls._triangles_to_trimesh(val.surface_mesh)
+        if not mesh.is_watertight:
+            raise ValidationError(
+                "The provided mesh is not watertight. "
+                "You can see which faces are broken using 'trimesh.repair.broken_faces'."
+            )
+        if not mesh.is_winding_consistent:
+            raise ValidationError(
+                "The provided mesh does not have consistent winding. "
+                "Consider using 'trimesh.repair.fix_winding'."
+            )
+        if not mesh.is_volume:
+            raise ValidationError("The provided mesh does not represent a valid (finite) volume.")
+        if not all(np.array(mesh.area_faces) > fp_eps):
+            raise ValidationError("The provided mesh has triangles with zero area.")
+        return val
+
+    @classmethod
+    def from_stl(
+        cls,
+        filename: str,
+        scale: float = 1.0,
+        origin: Tuple[float, float, float] = (0, 0, 0),
+        solid_index: int = None,
+        **kwargs,
+    ) -> Union[CustomSurfaceMeshGeometry, GeometryGroup]:
+        """Load a :class:`.CustomSurfaceMeshGeometry` directly from an STL file.
+        The ``solid_index`` parameter can be used to select a single solid from the file.
+        Otherwise, if the file contains a single solid, it will be loaded as a
+        :class:`.CustomSurfaceMeshGeometry`; if the file contains multiple solids,
+        they will all be loaded as a :class:`.GeometryGroup`.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the STL file containing the surface geometry mesh data..
+        scale : float
+            The length scale for the loaded geometry (um).
+        origin : Tuple[float, float, float]
+            The origin of the loaded geometry. Translates from (0,0,0) to this point.
+        solid_index : int
+            If set, read a single solid with this index from the file.
+
+        Returns
+        -------
+        Union[:class:`.CustomSurfaceMeshGeometry`, List[:class:`.GeometryGroup`]]
+            The geometry or geometries from the file.
+        """
+
+        def process_single(mesh: trimesh.Trimesh) -> CustomSurfaceMeshGeometry:
+            """Process a single trimesh using scale and origin."""
+            mesh.apply_scale(scale)
+            mesh.apply_translation(origin)
+            return cls.from_trimesh(mesh)
+
+        scene = trimesh.load(filename, **kwargs)
+        meshes = []
+        if isinstance(scene, trimesh.Trimesh):
+            meshes = [scene]
+        elif isinstance(scene.trimesh.Scene):
+            meshes = scene.dump()
+        else:
+            raise ValidationError(
+                "Invalid Trimesh type in file. Supported types are Trimesh and Scene."
```

Comment:
put 'Trimesh' and 'Scene' in quotes. and I'd consider adding `trimesh.` in front, like `'trimesh.Scene'`